pkg/core/resources/apis/meshservice/api/v1alpha1/helpers.go

```diff
@@ -43,15 +43,15 @@ func (m *MeshServiceResource) FindPortByName(name string) (Port, bool) {
 	return Port{}, false
 }
 
-func (m *MeshServiceResource) IsLocalMeshService(localZone string) bool {
+func (m *MeshServiceResource) IsLocalMeshService() bool {
```

Comment:
Why is this changing?

Reply:
Otherwise I would have to drag `localZone` through multiple layers to `GenerateClusters`, but in practice AFAIK this is how we detect if object is synced in other places.

Reply:
Ok maybe @lobkovilya but lgtm

Reply:
Yeah it makes sense, if the resource is synced from global it's going to have `kuma.io/origin: global` (even if initially it was synced from another zone)

Reply:
just to double check Ilya's concerns. This is output of 2.8
```
❯❯❯ kubectl get meshservices -A -o yaml
apiVersion: v1
items:
- apiVersion: kuma.io/v1alpha1
  kind: MeshService
  metadata:
    annotations:
      kuma.io/display-name: test-server
    creationTimestamp: "2024-10-18T13:21:12Z"
    generation: 4
    labels:
      k8s.kuma.io/namespace: msconnectivity
      kuma.io/env: kubernetes
      kuma.io/mesh: msconnectivity
      kuma.io/origin: global
      kuma.io/zone: kuma-1
    name: test-server-5d2zb94cwv8z4zff
    namespace: kuma-system
    resourceVersion: "990"
    uid: c3ae061d-34ff-4c00-a841-430f1acece06
  spec:
    identities:
    - type: ServiceTag
      value: test-server_msconnectivity_svc_80
    ports:
    - appProtocol: http
      port: 80
      targetPort: 80
    selector:
      dataplaneTags:
        app: test-server
        k8s.kuma.io/namespace: msconnectivity
  status:
    tls: {}
- apiVersion: kuma.io/v1alpha1
  kind: MeshService
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"kuma.io/v1alpha1","kind":"MeshService","metadata":{"annotations":{},"labels":{"kuma.io/env":"kubernetes","kuma.io/mesh":"msconnectivity","kuma.io/origin":"zone"},"name":"test-server","namespace":"msconnectivity"},"spec":{"ports":[{"appProtocol":"http","port":80,"targetPort":80}],"selector":{"dataplaneTags":{"app":"test-server","k8s.kuma.io/namespace":"msconnectivity"}}}}
      kuma.io/display-name: test-server
    creationTimestamp: "2024-10-18T13:21:24Z"
    generation: 1
    labels:
      k8s.kuma.io/namespace: msconnectivity
      kuma.io/env: kubernetes
      kuma.io/mesh: msconnectivity
      kuma.io/origin: zone
      kuma.io/zone: kuma-2
    name: test-server
    namespace: msconnectivity
    resourceVersion: "991"
    uid: a822d1a2-a9ee-42f0-804c-75e84b4d20a4
  spec:
    ports:
    - appProtocol: http
      port: 80
      targetPort: 80
    selector:
      dataplaneTags:
        app: test-server
        k8s.kuma.io/namespace: msconnectivity
  status:
    tls: {}
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```